src/main/java/core/basesyntax/strategy/operation/SupplyOperation.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.operation;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.Fruit;
+
+public class SupplyOperation implements OperationHandler {
+    private StorageDao storageDao = new StorageDaoImpl();
+
+    @Override
+    public void operationType(String fruit, int amount) {
+        if (storageDao.getFruit(fruit) != null) {
+            int newQuantity = storageDao.getFruit(fruit).getQuantity() + amount;
+            storageDao.update(fruit, newQuantity);
+        } else {
+            storageDao.addFruit(new Fruit(fruit, amount));
```

Comment:
Creating a new `Fruit` object directly in this method violates the Open/Closed Principle. Consider using a factory method or another design pattern to create `Fruit` instances, so you can extend the way they are created without modifying this class.